### PR TITLE
First implementation of using kustomize

### PR DESCRIPTION
**Pull request by victorpoluceno**
Very first draft of using kustomize instead of helm, a few things missing here, like executing it from the deploy script but the important is testing how to pass parameters around.

### CODE REVIEW

File: deploy/Makefile [hunk 1]
```diff
@@ -0,0 +1,5 @@
+example:
```
Comment (victorpoluceno) [inline deploy/Makefile:1]: This is useful to simulate how this is going to be done inside the deploy script and testing.

File: deploy/resources/base/ingress.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: ingress
+  annotations:
+    kubernetes.io/ingress.class: nginx
+spec:
+  tls:
+    - hosts:
+        - "id.domain"
```
Comment (victorpoluceno) [inline deploy/resources/base/ingress.yml:10]: These static values are replace by kustomize using the patches.

File: deploy/resources/base/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        value: 8880
+      - op: replace
+        path: /spec/ports/0/targetPort
+        value: 8880
+    target:
+      kind: Service
+  - patch: |-
+      - op: replace
+        path: /spec/tls/0/hosts/0
+        value: re-website-proxy-xxx.shared-prod.fih.io
```
Comment (victorpoluceno) [inline deploy/resources/base/kustomization.yaml:28]: These paths will be stored in a json file. Their values will be replaced by the real values by the deploy script. So, all parameters will be inject in the same way in the same place.

File: deploy/resources/example/deployment.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
+apiVersion: apps/v1
```
Comment (victorpoluceno) [inline deploy/resources/example/deployment.yml:1]: This is basically what they users need to do in order to override the base files.

File: deploy/Makefile [hunk 5]
```diff
@@ -0,0 +1,10 @@
+example-build:
+	mkdir -p resources/.kustomize/ resources/.kustomize/out && cp -R resources/example/* resources/.kustomize/
```
Comment (rlinstorres) [inline deploy/Makefile:2]: here you can run `mkdir -p resources/.kustomize/out`

File: deploy/resources/base/deployment.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        # Force all new release to recreate the pods #
+        rollme: "hash"
+      labels:
+        app.kubernetes.io/name: kube-review
+    spec:
+      containers:
+        - name: kube-review
+          securityContext:
+            null
+          image: "265409992602.dkr.ecr.eu-west-1.amazonaws.com/daedalus-proxy-main:44eddfd"
```
Comment (rlinstorres) [inline deploy/resources/base/deployment.yml:29]: here you have to change it to another thing, right?

File: deploy/resources/base/ingress.yml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+apiVersion: networking.k8s.io/v1beta1
```
Comment (rlinstorres) [inline deploy/resources/base/ingress.yml:1]: You need to change it to `networking.k8s.io/v1`, `v1beta1` will be deprated.

File: deploy/resources/base/kustomization.yaml [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        value: 8880
+      - op: replace
+        path: /spec/ports/0/targetPort
+        value: 8880
+    target:
+      kind: Service
+  - patch: |-
+      - op: replace
+        path: /spec/tls/0/hosts/0
+        value: re-website-proxy-xxx.shared-stg.fih.io
```
Comment (rlinstorres) [inline deploy/resources/base/kustomization.yaml:37]: Here, will you change it to another thing?

**Review by rlinstorres — COMMENTED** [review]
I left small comments.

File: deploy/resources/base/deployment.yml (hunk 6 above)
Comment (victorpoluceno) [reply deploy/resources/base/deployment.yml:29]: Yes, it's already getting replace, but I will add a more placeholder like value here.

File: deploy/resources/base/kustomization.yaml (hunk 8 above)
Comment (victorpoluceno) [reply deploy/resources/base/kustomization.yaml:37]: Yes, this is not yet implemented. But the patches are going to be stored in a json file and the values will be replaced during runtime by the deploy script.

File: deploy/Makefile [hunk 9]
```diff
@@ -0,0 +1,18 @@
+example-replace:
+	export NAMESPACE=re-website-proxy-xxx
+	export DOMAIN=shared-stg.fih.io
+	@for f in ./resources/.kustomize/base/patches/*.json; do \
+		cat $$f | envsubst; \
+	done
+
+example-build:
+	mkdir -p resources/.kustomize/out resources/.kustomize/base &&  \
```
Comment (rlinstorres) [inline deploy/Makefile:9]: you can use `mkdir -p resources/.kustomize/{out,base} &&` here

File: deploy/resources/base/ingress.yml (hunk 7 above)
Comment (rlinstorres) [reply deploy/resources/base/ingress.yml:1]: Don't forget to change it to `networking.k8s.io/v1`

**Review by rlinstorres — APPROVED** [review]
LGTM!
I guess we can do an internal demo for you to show us, how to use it, talk and so on.

**Review by gtonello — APPROVED** [review]
LGTM. Apparently it'll be simple for the users to understand the flow. And I agree with Rodrigo, a internal demo would be nice!

**Pull request closed without merging**